recreate_test.go

```diff
@@ -0,0 +1,206 @@
+//+build integration
+//+build scylla
+
+// Copyright (C) 2017 ScyllaDB
+
+package gocql
+
+import (
+	"io/ioutil"
+	"sort"
+	"strings"
+	"testing"
+	)
+func TestRecreateSchema(t *testing.T) {
+	session := createSessionFromCluster(createCluster(), t)
+	defer session.Close()
+
+	tcs := []struct {
+		Name     string
+		Keyspace string
+		Input    string
+		Golden   string
+	}{
+		{
+			Name:     "Keyspace",
+			Keyspace: "gocqlx_keyspace",
+			Input:    "testdata/recreate/keyspace.cql",
+			Golden:   "testdata/recreate/keyspace_golden.cql",
+		},
+		{
+			Name:     "Table",
+			Keyspace: "gocqlx_table",
+			Input:    "testdata/recreate/table.cql",
+			Golden:   "testdata/recreate/table_golden.cql",
+		},
+		{
+			Name:     "Materialized Views",
+			Keyspace: "gocqlx_mv",
+			Input:    "testdata/recreate/materialized_views.cql",
+			Golden:   "testdata/recreate/materialized_views_golden.cql",
+		},
+		{
+			Name:     "Index",
+			Keyspace: "gocqlx_idx",
+			Input:    "testdata/recreate/index.cql",
+			Golden:   "testdata/recreate/index_golden.cql",
+		},
+		{
+			Name:     "Secondary Index",
+			Keyspace: "gocqlx_sec_idx",
+			Input:    "testdata/recreate/secondary_index.cql",
+			Golden:   "testdata/recreate/secondary_index_golden.cql",
+		},
+		{
+			Name:     "UDT",
+			Keyspace: "gocqlx_udt",
+			Input:    "testdata/recreate/udt.cql",
+			Golden:   "testdata/recreate/udt_golden.cql",
+		},
+	}
+
+	for i := range tcs {
+		test := tcs[i]
+		t.Run(test.Name, func(t *testing.T) {
+			cleanup(t, session, test.Keyspace)
+
+			in, err := ioutil.ReadFile(test.Input)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			queries := trimQueries(strings.Split(string(in), ";"))
+			for _, q := range queries {
+				qr := session.Query(q, nil)
+				if err := qr.Exec(); err != nil {
+					t.Fatal("invalid input query", q, err)
+				}
+				qr.Release()
+			}
+
+			km, err := session.KeyspaceMetadata(test.Keyspace)
+			if err != nil {
+				t.Fatal("dump schema", err)
+			}
+			dump, err := km.Recreate()
+			if err != nil {
+				t.Fatal("recreate schema", err)
+			}
+
+			golden, err := ioutil.ReadFile(test.Golden)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			goldenQueries := sortQueries(strings.Split(string(golden), ";"))
+			dumpQueries := sortQueries(strings.Split(dump, ";"))
+
+			if len(goldenQueries) != len(dumpQueries) {
+				t.Errorf("Expected len(dumpQueries) to be %d, got %d", len(goldenQueries), len(dumpQueries))
+			}
+			// Compare with golden
+			for i, dq := range dumpQueries {
+				gq := goldenQueries[i]
+
+				if gq != dq {
+					t.Errorf("dumpQueries[%d] expected to be '%s', got '%s'", i, gq, dq)
+				}
+			}
+		})
+	}
+}
+
+func TestRecreatedSchemaCorrectness(t *testing.T) {
```

Comment:
Shouldn't this be part of the other test, `TestRecreateSchema` also checks correctness with Exec